### PR TITLE
Feature/return hidden observations api

### PR DESCRIPTION
**Pull request by tsubik**
- changing hidden filter for observation API
- remove not used observations CSV export

### CODE REVIEW

File: app/models/country.rb [hunk 1]
```diff
@@ -54,13 +54,9 @@ class Country < ApplicationRecord
...
 
-  scope :with_observations, (-> {
-    left_outer_joins(:observations).where.not(observations: { id: nil }).distinct
-  })
-
-  scope :with_active_observations, (-> {
-    joins(:observations).merge(Observation.active).where.not(observations: { id: nil }).distinct
-  })
+  scope :with_observations, ->(scope = Observation.all) {
+    joins(:observations).merge(scope).where.not(observations: { id: nil }).distinct
```
Comment (santostiago) [inline app/models/country.rb:58]: don't we want to keep the countries that have no observations?

File: app/resources/v1/observation_resource.rb [hunk 2]
```diff
@@ -40,8 +40,10 @@ class ObservationResource < JSONAPI::Resource
             :observation_report, :law, :operator, :subcategory,
             :is_active, :validation_status, :is_physical_place
 
-    filter :hidden, apply: ->(records, value, _options) {
-      records.unscope(where: :hidden).where(hidden: value)
+    filter :hidden, default: 'false', apply: ->(records, value, options) {
+      return records if value.include?('all')
```
Comment (santostiago) [inline app/resources/v1/observation_resource.rb:44]: will this work? Won't the default scope override it?

File: app/controllers/v1/observation_filters_controller.rb [hunk 3]
```diff
@@ -69,113 +42,8 @@ def tree
       render json: filters
     end
 
-    def csv
```
Comment (santostiago) [inline app/controllers/v1/observation_filters_controller.rb:72]: I can't remember where this was used. Did you check it?

**Review by santostiago — APPROVED** [review]

File: app/models/country.rb (hunk 1 above)
Comment (tsubik) [reply app/models/country.rb:58]: No, why? The scope is with_observations, I just changed the method to take with what kind of observations.

File: app/resources/v1/observation_resource.rb (hunk 2 above)
Comment (tsubik) [reply app/resources/v1/observation_resource.rb:44]: I removed default scope, and I'm using default filter (`false`). Anyway, I think the usage of `unscope` is some kind of code smell, or quick workaround and probably should be refactored.

File: app/controllers/v1/observation_filters_controller.rb (hunk 3 above)
Comment (tsubik) [reply app/controllers/v1/observation_filters_controller.rb:72]: I asked Sophie, she said to remove it.

Comment (tsubik) [reply app/controllers/v1/observation_filters_controller.rb:72]: They are not using any CSV download, and this method has comment pleading for refactor, so the best refactor is to remove code if not needed :D

**Pull request merged**